docs/celestia-architecture/adr-009-cat-pool.md

```diff
@@ -0,0 +1,96 @@
+
+## Alternative Approaches
+
+A few variations on the design were prototyped and tested. An early implementation experimented with just `SeenTx`s. All nodes would gossip `SeenTx` upon receiving a valid tx. Nodes would not relay received transactions to peers that had sent them a `SeenTx`. However, in many cases this would lead to a node sending a tx to a peer before it was able to receive the `SeenTx` that the node had just sent. Even with a higher priority, a large amount of duplication still occured.
+
+Another trick was tested which involved adding a `From` field to the `SeenTx`. Nodes receiving the `SeenTx` would use the `NodeID` in `From` to check if they were already connected to that peer and thus could expect a transaction from them soon instead of immediately issuing a `WantTx`. In large scale tests, this proved to be surprisingly less efficient. This might be because a `SeenTx` rarely arrives from another node before the initial sender has broadcast to everyone. It may also be because in the testnets, each node was only connected to 10 other nodes, decreasing the chance that the node was actually connected to the original sender. The `From` field also added an extra 40 bytes to the `SeenTx` message. In the chart below, this experiemnt is shown as CAT2.
+
+## Status
+
+Proposed
+
+## Consequences
+
+To validate its effectiveness, the protocol was benchmarked against existing mempool implementations. This was done under close-to-real network environments which used [testground](https://github.com/testground/testground) and the celestia-app binary (@ v0.11.0) to create 100 validator networks. The network would then be subjected to PFBs from 1100 light nodes at 4kb per transaction. The network followed 15 second blocktimes with a maximum block size of roughly 8MB (these were being filled). This was run for 10 minutes before being torn down. The collected data was aggregated across the 100 nodes and is as follows:
+
+| Version | Average Bandwidth | Standard Deviation | Finalized Bandwidth |
+|-----|-----|------|------|
+| v0 | 982.66MB/s | 113.91MB/s | 11MB/s |
+| v1 | 999.89MB/s | 133.24MB/s | 11MB/s |
+| v2 (CAT) | 98.90MB/s | 18.95MB/s | 11MB/s |
+| v2 (CAT2) | 110.28MB/s | 33.49MB/s | 11MB/s |
+
+> Finalized bandwidth is the amount of bytes finalized by consensus per second whereas the other measurements are per node.
+
+Rather than just expressing the difference in bytes, this can also be viewed by the factor of duplication (i.e. the amount of times a transaction is received by a node)
+
+| Version | v0 | v1 | v2 (CAT) | v2 (CAT2) |
+| --------|----|----|----------|-----------|
+| Duplication | 17.61x | 17.21x | 1.75x | 1.85x |
+
+
+This, of course, comes at the cost of additional message overhead and there comes a point where the transactions are small enough that the reduction in duplication doesn't outweigh the extra state messages.
```

Comment:
It could be interesting to take a closer look at the message propagation delay and the comparison of the block time in the CAT versus previous versions. However, if it takes a lot of time, it doesn't have to be done for this specific pull request.

Reply:
Yes I agree. It's something I'd like to be more adequately measured.